### PR TITLE
Implemented features such as folder structure, snippet completion, autocomplete basic syntaxes and error highlighting

### PR DESCRIPTION
**Pull request by Nagateja2402**
1. Implemented the Sentinel: Init Command so that it create a directory structure for sentinel policies with empty files.
2. Implemented snippet completion for 
	- func, rule, main, for, any, print, filter
3. Implemented variable hinting when the variables are declared.
4. Implemented error displaying to the user when they save the current sentinel file.

### CODE REVIEW

**Comment by Nagateja2402** [comment]
https://github.com/hashicorp/vscode-sentinel/assets/80318002/238b3cce-838d-4cd0-8794-3a048ebcbe9e

Added the demo video for the reference

File: build/download-syntax.ts [hunk 1]
```diff
@@ -29,13 +29,13 @@ async function run(info: ExtensionInfo) {
   const release = `v${info.syntaxVersion}`;
 
   const fileName = `${info.name}.tmGrammar.json`;
-  const url = `https://github.com/hashicorp/syntax/releases/download/${release}/${fileName}`;
+  const url = `http://localhost:8000/sentinel.tmGrammar.json`;
```
Comment (absolutelightning) [inline build/download-syntax.ts:32]: Should this be localhost? This needs to be removed.

**Review by absolutelightning — CHANGES_REQUESTED** [review]

**Review by glennsarti — CHANGES_REQUESTED** [review]
I've given this a very brief review and there are already some glaring errors here;

* Errant `console.log` messages and commented code which should be deleted.
* Copying from Microsoft Language Server examples and not editing them for this project, and keeping the license information.
* Generated Javascript files are committed instead of being ignored.
* Using shell binaries with platform logic instead of using the agnostic node FS library

I expect that there are many other things I've missed as it was only a cursory review.

However I have some much larger architecture questions;

* The use of a language server seems way over-complicated and instead should just use the inbuilt VSCode API to do this; VSCode already has native support for [snippets](https://code.visualstudio.com/api/language-extensions/snippet-guide), VSCode has an inbuilt [completion provider API](https://code.visualstudio.com/api/references/vscode-api#CompletionItemProvider), VSCode has a [Tasks API](https://code.visualstudio.com/api/references/vscode-api#tasks). There's doesn't appear to be the need for all the complexity and overhead of a language server at all

* I have issues with running a `sentinel apply` as a validation step. Sentinel apply doesn't setup any testing information and will fail on any well written policy. For example, consider the [Policies Registry](https://registry.terraform.io/browse/policies). Take the source of any of those policies and `sentinel apply` wil always fail because the expected imports are not available. Combined with the inability to turn off this feature and no feedback if the user doesn't have `sentinel` installed, this is a poor user experience.

At the very least it feels like `sentinel test` is a much better indication of success/failure and VSCode has a Tests API just for that.

**Comment by glennsarti** [comment]
Closing as stale

**Pull request closed without merging**